invisible_cities/io/mcinfo_io_test.py

```diff
@@ -0,0 +1,146 @@
+import os
+import numpy  as np
+import tables as tb
+
+from .  mcinfo_io import load_mchits
+from .  mcinfo_io import load_mcparticles
+from .  mcinfo_io import load_mcsensor_response
+from .  mcinfo_io import mc_info_writer
+from .  mcinfo_io import read_mcinfo_evt
+
+from .. reco.tbl_functions import get_mc_info
+
+from pytest import mark
+parametrize = mark.parametrize
+
+
+@mark.serial
+@parametrize('skipped_evt, out_filename',
+                  ((0, 'test_mcinfo_skip_evt0.h5'), (1, 'test_mcinfo_skip_evt1.h5')))
+def test_non_consecutive_events(config_tmpdir, ICDIR, krypton_MCRD_file, skipped_evt, out_filename):
+    filein  = os.path.join(ICDIR, 'database/test_data/', krypton_MCRD_file)
+    fileout = os.path.join(config_tmpdir, out_filename)
+
+    with tb.open_file(filein) as h5in:
+        with tb.open_file(fileout, 'w') as h5out:
+
+            mc_writer = mc_info_writer(h5out)
+            mc_extents = h5in.root.MC.extents
+            events_in = np.unique(h5in.root.MC.extents[:]['evt_number'])
+
+            mc_info = get_mc_info(h5in)
+
+            #Skip the desired event (there are only 3 in the file)
+            events_to_copy = [evt for evt in events_in if evt != skipped_evt]
+
+            for evt in events_to_copy:
+                mc_writer(mc_info, evt)
+
+            events_out = np.unique(h5out.root.MC.extents[:]['evt_number'])
+
+            np.testing.assert_array_equal(events_to_copy, events_out)
+
+
+@mark.serial
+@parametrize('file_to_check, evt_to_be_read',
+                  (('test_mcinfo_skip_evt0.h5', 1), ('test_mcinfo_skip_evt1.h5', 0),
+                   ('test_mcinfo_skip_evt1.h5', 2)))
+def test_output_non_consecutive_events(config_tmpdir, ICDIR, krypton_MCRD_file, file_to_check, evt_to_be_read):
+        filein  = os.path.join(ICDIR, 'database/test_data/', krypton_MCRD_file)
+        filecheck = os.path.join(config_tmpdir, file_to_check)
+
+        with tb.open_file(filein) as h5in:
+            with tb.open_file(filecheck) as h5filtered:
+                
+                mc_info = get_mc_info(h5in)
+                filtered_mc_info = get_mc_info(h5filtered)
+                # test the content of events to be sure that they are written
+                # correctly
+                hit_rows, particle_rows = read_mcinfo_evt(mc_info, evt_to_be_read)
+                filtered_hit_rows, filtered_particle_rows = read_mcinfo_evt(filtered_mc_info,
+                                                                            evt_to_be_read)
+
+                for hitr, filtered_hitr in zip(hit_rows, filtered_hit_rows):
+                    assert np.allclose(hitr['hit_position'], filtered_hitr['hit_position'])
+                    assert np.allclose(hitr['hit_time']    , filtered_hitr['hit_time'])
+                    assert np.allclose(hitr['hit_energy']  , filtered_hitr['hit_energy'])
+                    assert hitr['label']                  == filtered_hitr['label']
```

Comment:
can you align `hitr` with the one in the other lines while maintaining? Same for `partr`